components/DealDisplayFull.js

```diff
@@ -66,8 +69,37 @@ const DealDisplayFull = ({ setSelectedBusinessLocation }) => {
     if (businessInfo) {
       const deals = await fetchDealsForBusiness(businessInfo.id);
       console.log('Deals for the selected business:', deals);
+  
+      // Fetch additional business details based on business ID
+      const businessDetails = await fetchBusinessDetails(businessInfo.id);
```

Comment:
The business deals come with this call so we should be able to just call this and deal with both the deals and business info.